packages/studio-plugin/src/createStudioPlugin.ts

```diff
@@ -33,6 +35,24 @@ export default async function createStudioPlugin(
   const pathToUserProjectRoot = getProjectRoot(cliArgs);
 
   const studioConfig = await getStudioConfig(pathToUserProjectRoot, cliArgs);
+
+  let pagesDevelopmentPort: Promise<number | undefined>;
```

Comment:
I think you can avoid the `let` by doing something like this instead
```ts
const pagesDevPromise: Promise<void> =
  studioConfig.isPagesJSRepo ? startPagesDevelopmentServer() : Promise.resolve();
```
you could alternatively make a wrapper function or even pass isPagesJSRepo into startPagesDevelopmentServer (the latter is a bit weird from a high level)